### PR TITLE
Test on new elixir versions

### LINKED ISSUE #1154: Test failure on elixir >1.12 otp 24
```
2) test unknown imported modules returns error (Absinthe.Schema.Notation.ImportTest)
     test/absinthe/schema/notation/import_test.exs:255
     Expected exception Absinthe.Schema.Error but got ArgumentError (could not load module Test.Unknown due to reason :nofile)
     stacktrace:
       (elixir 1.12.2) lib/code.ex:1623: Code.ensure_compiled!/1
       (absinthe 1.7.0) lib/absinthe/phase/schema/type_imports.ex:70: Absinthe.Phase.Schema.TypeImports.ensure_compiled/1
       (absinthe 1.7.0) lib/absinthe/phase/schema/type_imports.ex:36: Absinthe.Phase.Schema.TypeImports.do_imports/3
       (absinthe 1.7.0) lib/absinthe/phase/schema/type_imports.ex:21: Absinthe.Phase.Schema.TypeImports.handle_imports/1
       (absinthe 1.7.0) lib/absinthe/blueprint/transform.ex:16: anonymous fn/3 in Absinthe.Blueprint.Transform.prewalk/2
       (absinthe 1.7.0) lib/absinthe/blueprint/transform.ex:109: Absinthe.Blueprint.Transform.walk/4
       (elixir 1.12.2) lib/enum.ex:1704: Enum."-map_reduce/3-lists^mapfoldl/2-0-"/3
       (absinthe 1.7.0) lib/absinthe/blueprint/transform.ex:139: anonymous fn/4 in Absinthe.Blueprint.Transform.walk_children/5
       (elixir 1.12.2) lib/enum.ex:2385: Enum."-reduce/3-lists^foldl/2-0-"/3
       (absinthe 1.7.0) lib/absinthe/blueprint/transform.ex:114: Absinthe.Blueprint.Transform.walk/4
       (absinthe 1.7.0) lib/absinthe/blueprint/transform.ex:15: Absinthe.Blueprint.Transform.prewalk/2
       (absinthe 1.7.0) lib/absinthe/phase/schema/type_imports.ex:10: Absinthe.Phase.Schema.TypeImports.run/2
       (absinthe 1.7.0) lib/absinthe/pipeline.ex:384: Absinthe.Pipeline.run_phase/3
       (absinthe 1.7.0) lib/absinthe/schema.ex:360: Absinthe.Schema.__after_compile__/2
       (elixir 1.12.2) src/elixir_module.erl:369: anonymous fn/5 in :elixir_module.expand_callback/6
       (elixir 1.12.2) src/elixir_module.erl:368: :elixir_module.expand_callback/6
       (stdlib 3.17) lists.erl:1267: :lists.foldl/3
       (elixir 1.12.2) src/elixir_module.erl:150: :elixir_module.compile/5
       (stdlib 3.17) erl_eval.erl:685: :erl_eval.do_apply/6
       (elixir 1.12.2) src/elixir.erl:280: :elixir.recur_eval/3
```

### PR DESCRIPTION
**Pull request by kdawgwilk**
This adds newer elixir and otp versions to testing matrix and fixes a failing test
Closes #1154

### CODE REVIEW

**Comment by maartenvanvliet** [comment]
There are some examples e.g. https://github.com/dashbitco/nimble_parsec/blob/2c3d6f0c058196d01ecb771e472a06f229a7e551/.github/workflows/ci.yml#L48 setting a var in the test matrix that will  only run the formatter on elixir 1.13

**Comment by kdawgwilk** [comment]
Alright, thanks for the tip @maartenvanvliet I got that figured out and now all tests are passing but there are a few weird cases of tests failing on specific elixir and otp versions, and not fealing the best on how I got those passing 😬

File: mix.lock [hunk 1]
```diff
@@ -13,6 +13,6 @@
...
   "makeup_graphql": {:hex, :makeup_graphql, "0.1.2", "81e2939aab6d2b81d39ee5d9e13fae02599e9ca6e1152e0eeed737a98a5f96aa", [:mix], [{:makeup, "~> 1.0", [hex: :makeup, repo: "hexpm", optional: false]}, {:nimble_parsec, "~> 1.1", [hex: :nimble_parsec, repo: "hexpm", optional: false]}], "hexpm", "3390ab04ba388d52a94bbe64ef62aa4d7923ceaffac43ec948f58f631440e8fb"},
   "mix_test_watch": {:hex, :mix_test_watch, "1.0.2", "34900184cbbbc6b6ed616ed3a8ea9b791f9fd2088419352a6d3200525637f785", [:mix], [{:file_system, "~> 0.2.1 or ~> 0.3", [hex: :file_system, repo: "hexpm", optional: false]}], "hexpm", "47ac558d8b06f684773972c6d04fcc15590abdb97aeb7666da19fcbfdc441a07"},
-  "nimble_parsec": {:hex, :nimble_parsec, "1.2.2", "811e32fb77aabb2b5b6196b21f76fe6ba8b6861c3d8c9eaeedbbf1f4cda627d1", [:mix], [], "hexpm", "dd3504559b0ddfeb7f55297557313fc05340120a037f981a4775b1c43e61d1b9"},
+  "nimble_parsec": {:hex, :nimble_parsec, "1.2.2", "b99ca56bbce410e9d5ee4f9155a212e942e224e259c7ebbf8f2c86ac21d4fa3c", [:mix], [], "hexpm", "98d51bd64d5f6a2a9c6bb7586ee8129e27dfaab1140b5a4753f24dac0ba27d2f"},
```
Comment (kdawgwilk) [inline mix.lock:16]: I am not sure how this hash was out of date but a fresh checkout of master requires running `mix deps.get` and updates the lock file

File: test/absinthe/phase/parse_test.exs [hunk 2]
```diff
@@ -82,14 +82,27 @@ defmodule Absinthe.Phase.ParseTest do
...
-               phase: Absinthe.Phase.Parse
-             }
-           ] == bp.execution.validation_errors
+    otp = :erlang.system_info(:otp_release)
+
+    if otp == '24' do
+      assert [
+               %Absinthe.Phase.Error{
+                 extra: %{},
+                 locations: [%{column: 3, line: 4}],
```
Comment (kdawgwilk) [inline test/absinthe/phase/parse_test.exs:91]: weird behavior difference where otp 24 now has column info

File: lib/absinthe/phase/schema/type_imports.ex [hunk 3]
```diff
@@ -71,6 +71,13 @@ defmodule Absinthe.Phase.Schema.TypeImports do
     else
       Code.ensure_compiled(module)
     end
+  catch
+    # Code.ensure_compiled! in Elixir >1.12 raises an ArgumentError if it is unable to find the module with message similar to
+    # "could not load module <module> due to reason <reason>"
+    # where reason is an atom :embedded | :badfile | :nofile | :on_load_failure | :unavailable
+    _, %ArgumentError{message: message} ->
+      reason = message |> String.split(":") |> List.last()
```
Comment (kdawgwilk) [inline lib/absinthe/phase/schema/type_imports.ex:79]: Not a fan of this as I am sure its pretty brittle, open to suggestions

File: test/absinthe/phase/parse_test.exs (hunk 2 above)
Comment (benwilson512) [reply test/absinthe/phase/parse_test.exs:91]: Yeah this is expected, this is a fine way to handle it.

**Review by benwilson512 — APPROVED** [review]
This looks good, thanks for working on this!

**Pull request merged**